base/src/main/java/io/spine/code/js/DirectoryReference.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.code.js;
+
+import com.google.common.base.Splitter;
+import io.spine.value.StringTypeValue;
+
+import java.util.List;
+
+import static io.spine.util.Preconditions2.checkNotEmptyOrBlank;
+
+/**
+ * A reference to a directory.
+ *
+ * <p>May include parent directories separated by {@linkplain FileReference#separator() slashes},
+ * e.g. {@code root/sub}.
+ */
+public final class DirectoryReference extends StringTypeValue {
+
+    private static final long serialVersionUID = 0L;
+
+    private DirectoryReference(String value) {
+        super(value);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param value
+     *         the value of the reference
+     * @return a new instance
+     */
+    public static DirectoryReference of(String value) {
+        checkNotEmptyOrBlank(value);
+        return new DirectoryReference(value);
+    }
+
+    /**
+     * Tells if this directory is the parent for specified directory.
+     *
+     * @param target
+     *         the directory to check
+     * @return {@code true} if specified directory is a child of this one
+     */
+    public boolean isParentFor(DirectoryReference target) {
+        boolean sameBeginning = target.value()
+                                      .startsWith(value());
+        boolean result = sameBeginning && target.length() > length();
```

Comment:
So `foobar` is the parent of `foobarbla/sub`?

Reply:
@dmytro-grankin Actually, my example is wrong. But consider having `parent` and `parentchild`. They aren't in any relationship, but the condition is true.